src/main.cpp

```diff
@@ -6502,9 +6509,13 @@ bool SendMessages(CNode* pto, CConnman& connman)
     const Consensus::Params& consensusParams = Params().GetConsensus();
     {
         // Don't send anything until we get its version message
-        if (pto->nVersion == 0)
+        if (!pto->fSuccessfullyConnected || pto->fDisconnect)
```

Comment:
@dcousens the version message sets fSuccessfullyConnected, but I guess that's not as clear after this change.

How about "don't send anything until we're successfully connected and ok with the peer's version" ?